src/NUnitFramework/framework/Constraints/UniqueItemsConstraint.cs

```diff
@@ -52,97 +56,172 @@ public override string Description
         /// <returns></returns>
         protected override bool Matches(IEnumerable actual)
         {
-            return TryFastAlgorithm(actual) ?? OriginalAlgorithm(actual);
+            var nonUniqueItems = GetNonUniqueItems(actual);
+            return nonUniqueItems.Count == 0;
+        }
+
+        /// <inheritdoc />
+        public override ConstraintResult ApplyTo<TActual>(TActual actual)
+        {
+            IEnumerable enumerable = ConstraintUtils.RequireActual<IEnumerable>(actual, nameof(actual));
+
+            var nonUniqueItems = GetNonUniqueItems(enumerable);
+
+            return new UniqueItemsConstraintResult(this, actual, nonUniqueItems);
         }
 
-        private bool OriginalAlgorithm(IEnumerable actual)
+        private ICollection OriginalAlgorithm(IEnumerable actual)
         {
-            var list = new List<object>();
+#if !NET35
```

Comment:
The `IComparable` fastpath uses a `SortedList<T>`, which doesn't exist in NET35

Reply:
This would make a nice comment on the same line.